add documentation and cleanup existing documentation (#370)
tijlleenders/ZinZen-scheduler: documentation/functional/Concepts.md

## Concepts in the ZinZen&reg; scheduling algorithm
To create a ubiquitous language for talking about the algorithm, all concepts used in the algorithm are defined below.

### 1) Goals
A goal is the most important concept in ZinZen&reg;. A goal is a description of something 

> you want to get done, e.g. 'walk 4 hours'.

Reviewer: ```suggestion
you want to get done, e.g. 'walk 4 hours'. Goals come from the frontend/UI and are specified by the user.
```

Goals can be modified with restrictions and repeats, e.g. 'walk 4 hours every week between 6p.m. and 10p.m from Sep. 1st until Oct. 1st'

### 2) Steps
Steps are concrete steps to reach a goal with all it's restrictions and modifiers.

They contain a start time, a deadline and a duration.

The example above ('walk 4 hours every week between 6p.m. and 10p.m from Sep. 1st until Oct. 1st')

> can be split into 1 step per relevant week, with a start time on the first time of the week, a deadline on the last

Reviewer: ```suggestion
can be split into mulitple Steps of a week each, beginning with the start date-time and incrementing in 'steps' of 7 day intervals. The last Step can be shortened if the deadline is reached.
```

> can be split into mulitple Steps of a week each, beginning with the start date-time and incrementing in 'steps' of 7 day intervals. The last Step can be shortened if the deadline is reached.

Reviewer: ```suggestion
can be split into multiple Steps of a week each, beginning with the start date-time and incrementing in 'steps' of 7 day intervals. The last Step can be shortened if the deadline is reached.
```
